src/reducers/counter.js

```diff
@@ -3,29 +3,52 @@
 import { Actions } from "../actions/counter";
 
 const counter = {
-  count: 0,
-  count2: 0
+  count: 0
 };
 
-export const CounterReducer = (state = counter, action) => {
+const counterCount = {
+  counterList: []
+};
+
+export const counterCountReducer = (state = counterCount, action) => {
   switch (action.type) {
-    case Actions.Increment:
-      return Object.assign({}, state, {
-        count: state.count + 1
-      });
-    case Actions.Decrement:
+    case Actions.newCounter: {
+      const newCounterList = [...state.counterList, { count: 0 }];
+      //count:0 ??
       return Object.assign({}, state, {
-        count: state.count - 1
+        counterList: newCounterList
       });
+    }
+    case Actions.Increment: {
+      const counterList2 = state.counterList.map(counter => counter);
+      const newcount = counterList2[action.key].count + 1;
+      counterList2[action.key].count = newcount;
+      // const counterList2 = state.counterList.map(
+      //   (counter, index) => counter + (index === action.key)
+      //   // index = {key}
+      // );
+
+      return Object.assign({ counterList: counterList2 }, state);
+    }
+
+    // const counterList2 = state.counterList.map(
```

Comment:
I'm guessing this is left in because you had trouble figuring out map? No worries let's talk about it today. Map, filter and reduce can take a while to get your head around.